sympy/solvers/tests/test_solvers.py

```diff
@@ -204,6 +204,9 @@ def test_solve_args():
     # ordering should be canonical, fastest to order by keys instead
     # of by size
     assert list(solve((y - 1, x - sqrt(3)*z)).keys()) == [x, y]
+    # as set always returns as symbols, set even if no solution
+    assert solve([x - 1, x], (y, x), set=True) == ([y, x], set())
+    assert solve([x - 1, x], {y, x}, set=True) == ([x, y], set())
```

Comment:
Since `set` order is not deterministic is this guaranteed to pick the order `x, y`? Is that handled somewhere?

Reply:
Yes, when symbols are given in a well defined way that order is retained. When given as a set, they are always put in canonical order.